crates/nu-engine/src/eval.rs

```diff
@@ -1015,9 +1015,7 @@ pub fn eval_block(
     let num_pipelines = block.len();
 
     for (pipeline_idx, pipeline) in block.pipelines.iter().enumerate() {
-        let mut i = 0;
-
-        while i < pipeline.elements.len() {
+        for i in 0..pipeline.elements.len() {
```

Comment:
Don't like the `i` for such a nest-fest.

As the whole thing cares about `pipeline.elements[i]` and checks `pipeline.elements[i+1]` I am wondering if here `elements.iter().peekable()` could simplify things.

Reply:
I already had that, but was unsure which one was nicer :D
Will go for the peekable then

Reply:
Done

Reply:
Lightning fast!

While it doesn't really save on line-real-estate, I think that this has less chances to mess things up.

Reply:
> While it doesn't really save on line-real-estate, I think that this has less chances to mess things up.

Agreed.